cmd/shutdown_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"database/sql"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/connect"
+)
+
+func Test_gracefulShutdown_Successful(t *testing.T) {
+	signal := make(chan bool)
+
+	client, err := api.NewClient(api.DefaultConfig())
```

Comment:
It would be great to use io.Closer mock instead of consul client or db connection

Reply:
fixed